cobigen/core-artifact-retriever/src/test/java/com/devonfw/cobigen/retriever/MavenProxyTest.java

```diff
@@ -0,0 +1,120 @@
+package com.devonfw.cobigen.retriever;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.devonfw.cobigen.api.exception.CobiGenRuntimeException;
+import com.devonfw.cobigen.retriever.settings.MavenProxy;
+import com.devonfw.cobigen.retriever.settings.MavenSettings;
+import com.devonfw.cobigen.retriever.settings.to.model.MavenSettingsModel;
+import com.devonfw.cobigen.retriever.settings.to.model.MavenSettingsProxyModel;
+import com.devonfw.cobigen.retriever.settings.to.model.MavenSettingsRepositoryModel;
+
+/**
+ * Test class for MavenProxy
+ */
+public class MavenProxyTest {
+
+  /** Test data root path */
+  private static final String testdataRoot = "src/test/resources/testdata/unittest/MavenProxyTest";
+
+  /** Repositories on which the tests are performed on */
+  private static List<MavenSettingsRepositoryModel> repositoriesList;
+
+  /** Proxies, on which the tests are performed on */
+  private static List<MavenSettingsProxyModel> proxiesList;
+
+  /**
+   * Used to initialize data needed for the tests
+   */
+  @BeforeClass
+  public static void setUpClass() {
+
+    MavenSettingsModel model;
+    String content;
+    try {
+      content = new String(Files.readAllBytes(Paths.get(testdataRoot).resolve("settings.xml")));
+    } catch (IOException e) {
+      throw new CobiGenRuntimeException("Unable to read test settings.xml", e);
+    }
+    model = MavenSettings.generateMavenSettingsModel(content);
+
+    repositoriesList = new ArrayList<>();
+    proxiesList = new ArrayList<>();
+
+    // Get all the repositories for the test
+    repositoriesList = new ArrayList<>();
+    repositoriesList.addAll(model.getProfiles().getProfileList().get(0).getRepositories().getRepositoryList());
+    repositoriesList.addAll(model.getProfiles().getProfileList().get(1).getRepositories().getRepositoryList());
+    repositoriesList.addAll(model.getProfiles().getProfileList().get(2).getRepositories().getRepositoryList());
+
+    // Get all the mirrors for the test
+    proxiesList = new ArrayList<>();
+    proxiesList = new LinkedList<>();
+    proxiesList.addAll(model.getProxies().getProxyList());
```

Comment:
This looks like redundant code from MavenMirrorTest.

Reply:
The repositories are needed in order to get the repositories which are using the proxy